### PR TITLE
Add sign-latest-release.py to the electron scripts

### PR DESCRIPTION
**Pull request by vxgmichel**
A python script to help automate the signing of the windows release using the latest pre-built artefact.

### CODE REVIEW

File: client/electron/scripts/sign-latest-release.py [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+        "X-GitHub-Api-Version": "2022-11-28",
+    }
+    return urllib3.request("GET", url, headers=headers, preload_content=False)  # type: ignore
+
+
+def get_artifact(
+    base_url: str, token: str, version: None | str = None
+) -> tuple[str, int, str, str, str]:
+    url = f"{base_url}/actions/artifacts"
+    artefacts = get(url, token).json()["artifacts"]  # type: ignore
```
Comment (mmmarcos) [inline client/electron/scripts/sign-latest-release.py:56]: Pedantic mode: you are mixing "art**i**facts" (US spelling) and "art**e**facts" (UK spelling). I prefer artifact, but [both are accepted](https://english.stackexchange.com/a/37904). BTW GitHub calls them "assets" :sweat_smile:

File: client/electron/scripts/sign-latest-release.py [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(description="Sign the latest release artifact.")
+    parser.add_argument(
+        "--version",
+        type=str,
+        help="Specify the version of the artifact to sign (e.g., v1.2.3). If not provided, the latest version will be used.",
+    )
+    args = parser.parse_args()
+    version = args.version if args.version.startswith("v") else f"v{args.version}"
+    main(version)
```
Comment (mmmarcos) [inline client/electron/scripts/sign-latest-release.py:137]: Sure about "If not provided, the latest version will be used"?

I didn't run the script, but it seems to me that if not provided, `version` is set to `"v"` on line 136, which would result in Artifact not found error.

File: client/electron/scripts/sign-latest-release.py [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+This script downloads the latest pre-built artifact from the GitHub repository,
+unzips it, checks the version, and create a signed windows installer using the
+embedded signing script.
+
+
+Requirements:
+- Python 3.9+
+- gh (GitHub CLI) installed and authenticated
+- tqdm installed (pip install tqdm)
+- urllib3>=2 installed (pip install "urllib3>=2")
```
Comment (mmmarcos) [inline client/electron/scripts/sign-latest-release.py:15]: Also the sign script, isn't it? held by project admins? 
```suggestion
- urllib3>=2 installed (pip install "urllib3>=2")
- The sign script (sign-windows-package.cmd) held by project administrators
```

File: client/electron/scripts/sign-latest-release.py (hunk 1 above)
Comment (vxgmichel) [reply client/electron/scripts/sign-latest-release.py:56]: Github uses for its API btw: https://docs.github.com/en/rest/actions/artifacts

File: client/electron/scripts/sign-latest-release.py (hunk 3 above)
Comment (vxgmichel) [reply client/electron/scripts/sign-latest-release.py:15]: No, the sign script is part of the artifact

File: client/electron/scripts/sign-latest-release.py (hunk 2 above)
Comment (vxgmichel) [reply client/electron/scripts/sign-latest-release.py:137]: `args.version` would be `None` so that would be an attribute error, good catch

Comment (vxgmichel) [reply client/electron/scripts/sign-latest-release.py]: b7a861c8fd56a84435a88315563c128a4797b220

File: client/electron/scripts/sign-latest-release.py (hunk 1 above)
Comment (vxgmichel) [reply client/electron/scripts/sign-latest-release.py]: b7a861c8fd56a84435a88315563c128a4797b220

**Review by mmmarcos — APPROVED** [review]

**Pull request merged**